From c00409b1d398a073598e9119c8e2feb9ab07dc04 Mon Sep 17 00:00:00 2001
From: Galih Tama <galpt@v.recipes>
Date: Tue, 4 Aug 2026 01:12:15 +0700
Subject: [PATCH 35/36] v4.8-gpu: include linux/cpuset.h from sched.h for
 strict-GCC builds

sched_rt_handler() calls rebuild_sched_domains(), declared only in
include/linux/cpuset.h.  With GCC 14+ (implicit-function-declaration is
a hard error in gnu11) the vanilla 6.18 build fails rt.o with an
implicit declaration, because no include in the rt.c chain reaches
cpuset.h.  Upstream added this include to sched.h between 6.18 and 7.1;
mirror that fix here so the 6.18 tree compiles with the reference
toolchain.

Signed-off-by: Galih Tama <galpt@v.recipes>
---
 kernel/sched/sched.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kernel/sched/sched.h b/kernel/sched/sched.h
index b8ef4a1f2..8575f8438 100644
--- a/kernel/sched/sched.h
+++ b/kernel/sched/sched.h
@@ -31,6 +31,7 @@
 #include <linux/context_tracking.h>
 #include <linux/cpufreq.h>
 #include <linux/cpumask_api.h>
+#include <linux/cpuset.h>
 #include <linux/ctype.h>
 #include <linux/file.h>
 #include <linux/fs_api.h>
-- 
2.55.0

